### PR TITLE
Add Java tour examples

### PR DESCRIPTION
**Pull request by pcholakov**

### CODE REVIEW

File: java/src/main/java/dev/restate/tour/part2/TicketService.java [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+        var status = ctx.get(STATE_KEY).orElse(TicketStatus.Available);
+
+        if (!status.equals(TicketStatus.Sold)) {
+            ctx.clear(STATE_KEY);
+        }
+
+        ctx.sideEffect(CoreSerdes.BOOLEAN, () -> {
+            var result = PaymentClient.get().call("tx-id", 500);
+            if (result) {
+                throw new io.grpc.StatusRuntimeException(Status.UNKNOWN);
```
Comment (slinkydeveloper) [inline java/src/main/java/dev/restate/tour/part2/TicketService.java:42]: Was this supposed to throw `TerminalException`?

File: java/src/main/java/dev/restate/tour/auxiliary/PaymentClient.java [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+
+    public boolean failingCall(String idempotencyKey, double amount) {
+        if (i >= 2) {
+            System.out.println("Payment call succeeded for idempotency key " + idempotencyKey + " and amount " + amount);
+            i = 0;
+            return true;
+        } else {
+            System.out.println("Payment call failed for idempotency key " + idempotencyKey + " and amount " + amount + ". Retrying...");
+            i = i + 1;
+            throw new Error("Payment call failed");
```
Comment (slinkydeveloper) [inline java/src/main/java/dev/restate/tour/auxiliary/PaymentClient.java:24]: `Error`?! Maybe `IllegalStateException` or some other exception?

File: java/src/main/resources/log4j2.properties [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+
+# Console appender configuration
+appender.console.type = Console
+appender.console.name = consoleLogger
+appender.console.layout.type = PatternLayout
+appender.console.layout.pattern = %d{yyyy-MM-dd HH:mm:ss} [%tn] %-5p %c{1}:%L - %m%n
+
+# Restate logs to debug level
+logger.app.name = dev.restate
+logger.app.level = debug
```
Comment (slinkydeveloper) [inline java/src/main/resources/log4j2.properties:12]: Perhaps `info` is enough?

File: java/src/main/resources/log4j2.properties [hunk 4]
```diff
@@ -0,0 +1,18 @@
+# Set to debug or trace if log4j initialization is failing
+status = warn
+
+# Console appender configuration
+appender.console.type = Console
+appender.console.name = consoleLogger
+appender.console.layout.type = PatternLayout
+appender.console.layout.pattern = %d{yyyy-MM-dd HH:mm:ss} [%tn] %-5p %c{1}:%L - %m%n
```
Comment (slinkydeveloper) [inline java/src/main/resources/log4j2.properties:8]: Perhaps let's update this with the pattern containing more info? https://github.com/restatedev/sdk-java#logging

**Review by slinkydeveloper — APPROVED** [review]

File: java/src/main/resources/log4j2.properties (hunk 4 above)
Comment (gvdongen) [reply java/src/main/resources/log4j2.properties:8]: hmm, I saw this a bit late... If I change this, I need to change all the log output in the tour... Can I leave it for now?

**Review by pcholakov — COMMENTED** [review]
Latest changes look good, @gvdongen!

File: java/src/main/resources/log4j2.properties (hunk 4 above)
Comment (slinkydeveloper) [reply java/src/main/resources/log4j2.properties:8]: we can leave it for the time being.

**Pull request merged**